src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferLengthCountIsEmptyOverAny.Fixer.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    public abstract class PreferLengthCountIsEmptyOverAnyFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(
+            PreferLengthCountIsEmptyOverAnyAnalyzer.LengthId,
+            PreferLengthCountIsEmptyOverAnyAnalyzer.CountId,
+            PreferLengthCountIsEmptyOverAnyAnalyzer.IsEmptyId
+        );
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root?.FindNode(context.Span);
+            if (node is null)
+            {
+                return;
+            }
```

Comment:
I think the `Microsoft.CodeAnalysis` version used in this repo has non-nullable return for `GetSyntaxRootAsync`, so null check is unnecessary.

```suggestion
            var node = root.FindNode(context.Span);
```

Later, when we move to a newer version where `root` will become nullable, it's more likely we'll introduce a `GetRequiredSyntaxRootAsync` extension.